ci-operator/step-registry/ibmcloud/provision/vpc/ibmcloud-provision-vpc-commands.sh

```diff
@@ -173,11 +184,49 @@ function attach_public_gateway_to_subnet() {
     "${IBMCLOUD_CLI}" is subnet-update ${subnetName} --vpc ${vpcName} --pgw ${pgwName} || return 1
 }
 
+# for verify case  OCPBUGS-36236 [IBMCloud] install only checks first set of subnets (no pagination support)
+# for verify case OCPBUGS-36185[IBMCloud] MAPI only checks first set of subnets (no pagination support)
+function checkUsedSubnets() {
+    local used_subnets_file="$1" rg="$2"
+    local used_subnets TOKEN next_start subnets2 hasCp hasCompute
+    readarray -t used_subnets < <(yq-go r ${used_subnets_file} | awk '{print $2}')
+
+    TOKEN=$("${IBMCLOUD_CLI}" iam oauth-tokens | awk '{print $4}')
+    rg_id=$("${IBMCLOUD_CLI}" resource group $rg --output json | jq -r '.[0]|.id')
+    filter="resource_group.id=$rg_id&generation=2&version=2024-11-05"
+    echo "filter subnets with $filter"
+    next_start=$(curl -s -X GET "https://$region.iaas.cloud.ibm.com/v1/subnets?${filter}" -H "Authorization: Bearer $TOKEN" | jq -r '.next.href' | awk -F 'start=' '{print $2}')
+    subnets2=$(curl -s -X GET "https://$region.iaas.cloud.ibm.com/v1/subnets?${filter}&start=$next_start" -H "Authorization: Bearer $TOKEN" | jq -r '.subnets[]|.name')
+    echo "second page subnets: " "${subnets2[@]}"
+    hasCp=false
+    hasCompute=false
+    for subnet in "${subnets2[@]}"; do
+        if echo "${used_subnets[@]}" | grep -q "$subnet"; then
+            echo "Found: $subnet in the second page"
+            if [[ $subnet == *"control-plane"* ]]; then
+                hasCp=true                
+            elif [[ $subnet == *"compute"* ]]; then
```

Comment:
To check if compute subnets are used or not, we should use passed `used_subnets_file` to determinate, but not detect subnet name.